streams/src/main/java/org/apache/kafka/streams/state/internals/OffsetCheckpoint.java

```
@@ -114,67 +119,66 @@ private void writeEntry(BufferedWriter writer, TopicPartition part, long offset)
+                            final long offset = Long.parseLong(pieces[2]);
                             offsets.put(new TopicPartition(topic, partition), offset);
                             line = reader.readLine();
                         }
-                        if (offsets.size() != expectedSize)
-                            throw new IOException(String.format("Expected %d entries but found only %d",
-                                expectedSize,
-                                offsets.size()));
+                        if (offsets.size() != expectedSize) {
+                            throw new IOException(
+                                String.format("Expected %d entries but found only %d", expectedSize, offsets.size()));
+                        }
                         return offsets;
 
                     default:
                         throw new IllegalArgumentException("Unknown offset checkpoint version: " + version);
                 }
-            } finally {
-                reader.close();
+            } catch (final FileNotFoundException e) {
+                return Collections.emptyMap();
             }
         }
     }
 
     /**
      * @throws IOException if file read ended prematurely
      */
-    private int readInt(BufferedReader reader) throws IOException {
-        String line = reader.readLine();
-        if (line == null)
+    private int readInt(final BufferedReader reader) throws IOException {
+        final String line = reader.readLine();
+        if (line == null) {
             throw new EOFException("File ended prematurely.");
+        }
         return Integer.parseInt(line);
     }
 
     /**
      * @throws IOException if there is any IO exception during delete
      */
     public void delete() throws IOException {
-        Files.delete(file.toPath());
+        try {
+            Files.delete(file.toPath());
+        } catch (final NoSuchFileException e) { }
```

This change causes a findBugs warning causing the build to fail. It seems suspect to swallow the exception in this way as well.

---

It's correct to swallow this exception... Alternatively, we could check if the File exists (right now we just call "delete" blindly) before trying to delete it -- to prevent the exception in the first place. I see now reason to log this or do anything else. What else could I do to fix findBugs? Would a comment do?

---

If the file not existing is an expected scenario, you should use Files.deleteIfNotExists. Swallowing the exception does capture the intent and is verbose while checking if the file exists as a separate step is unsafe if there is concurrency.